### ISSUE TITLE
Move `Gate.decompose()` methods to transpiler

### ISSUE THREAD

**Issue by stavros11** [qiboteam/qibo#1040]
For when we move the transpilers from qibolab to qibo, it would be good to have a unified approach of gate transformations.

_Originally posted by @stavros11 in https://github.com/qiboteam/qibo/pull/1035#pullrequestreview-1670424696_

**Comment by renatomello** [comment]
I do not know how is the situation with the transpilers right now, but it is good to point out that decompositions are generally not unique. So we should also take that into account and have multiple decompositions at least for the most important gates (e.g. $U_{3}$)

**Comment by stavros11** [comment]
> I do not know how is the situation with the transpilers right now, but it is good to point out that decompositions are generally not unique. So we should also take that into account and have multiple decompositions at least for the most important gates (e.g. U3)

Indeed, in the current qibolab transpiler we have a [`GateDecompositions`](https://github.com/qiboteam/qibolab/blob/8746607a2e1435d9525b17c4d80c30c93fdd7315/src/qibolab/transpilers/unroller.py#L176) object which holds the decomposition of each gate. The idea is that you can have multiple instances of this object holding different decompositions depending on the application.

**Comment by renatomello** [comment]
@Simone-Bordoni @stavros11 now that the transpiler was moved to `qibo`, could we take a look at this again?
